pkg/parser/subnetparser.go

```diff
@@ -0,0 +1,117 @@
+package parser
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/Nordix/eno/pkg/cni"
+	"net"
+
+	enov1alpha1 "github.com/Nordix/eno/api/v1alpha1"
+	"github.com/go-logr/logr"
+)
+
+// SubnetParser instance
+type SubnetParser struct {
+	subnetResource *enov1alpha1.Subnet
+	log            logr.Logger
+}
+
+func NewSubnetParser(subnetObj *enov1alpha1.Subnet, logger logr.Logger) *SubnetParser {
+	return &SubnetParser{subnetResource: subnetObj,
+		log: logger}
+}
+
+
+func (sp *SubnetParser) ValidateSubnet() error {
+	ipStr := sp.subnetResource.Spec.Address
+	ipAddr := net.ParseIP(ipStr)
+	if ipAddr == nil {
+		err := fmt.Errorf("Invalid IP: %s", ipStr)
+		sp.log.Error(err, "")
+		return err
+	}
+	ipType := sp.subnetResource.Spec.Type
+	if ipType != "v4" && ipType != "v6" {
```

Comment:
I do not think we need this check if we have the Enum comment in the Subnet_types.go file

Reply:
ok i will remove this validation.